core/src/main/java/org/apache/accumulo/core/clientImpl/InstanceOperationsImpl.java

```diff
@@ -214,38 +222,46 @@ public Map<String,String> getSystemProperties()
   }
 
   @Override
+  @Deprecated(since = "4.0.0")
   public List<String> getManagerLocations() {
-    return context.getManagerLocations();
+
+    Set<ServerId> managers = getServers(ServerTypeName.MANAGER);
+    if (managers == null || managers.isEmpty()) {
+      return List.of();
+    } else {
+      return List.of(managers.iterator().next().toHostPortString());
```

Comment:
Yes, currently only one manager is supported, the one that holds the lock. If we move to multiple managers, then this will have to change. The zookeeper path for the Manager lock will have to change too.